matrix-synapse: 1.21.2 -> 1.22.0

Ma27 opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions
-->

###### Motivation for this change

https://github.com/matrix-org/synapse/releases/tag/v1.22.0

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Ma27 committed:
matrix-synapse: 1.21.2 -> 1.22.0

https://github.com/matrix-org/synapse/releases/tag/v1.22.0

SuperSandro2000 reviewed (approved):
Builds on darwin with #101319 merged.

mweinelt commented on pkgs/servers/matrix-synapse/default.nix:
```diff
@@ -53,6 +53,7 @@ buildPythonApplication rec {
     typing-extensions
     authlib
     pyjwt
+    hiredis
```
Redis integration is only used when the process is split up into multiple workers. Is that something you plan to do?

Ma27 replied:
Without adding this dependency, the build would've failed locally. OTOH I don't see a change in the relevant files, so I may need to recheck that.

Ma27 replied:
So you're right, this is strictly speaking an optional dependency. Should we exclude it for now or leave it for folks who want to do some work to get workers running?

mweinelt replied:
I don't mind this at all.

Ekleog replied:
The best would maybe be to add a `enableRedis ? false` flag to the derivation if it doesn't work with the nixos module, and to add this as `lib.optional`?

Ma27 replied:
Fixed %)

mweinelt commented:
Result of `nixpkgs-review pr 101853` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>matrix-synapse</li>
  </ul>
</details>

Ekleog reviewed (approved):
LGTM, just had a comment about potential improvements for the optional dependency but IMO this is good enough to land as is if you don't feel like doing it now

Ma27 committed:
matrix-synapse: make dependency for `hiredis` optional

This is only needed for replication which isn't even supported by the
current NixOS module.

mweinelt commented:
Result of `nixpkgs-review pr 101853` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>matrix-synapse</li>
  </ul>
</details>

Ma27 commented:
@mweinelt do you also take care of the backport? %)

mweinelt commented:
Feel free to do it, I'm busy with #101968 right now.

Ma27 commented:
20.03: e39c1f4ef17d36ce04fb75bd36f2e62b44ce0028, 504f993df9a5ca63317fe9c859df19daad30aa14
20.09: bf8631151e80717aa323d2b6e0ae36da175d8d25, 86777b5e659a13193312582e180093a9952eff0c
